auth/store.go

```diff
@@ -781,6 +800,18 @@ func (as *authStore) RoleGrantPermission(r *pb.AuthRoleGrantPermissionRequest) (
 	})
 
 	if idx < len(role.KeyPermission) && bytes.Equal(role.KeyPermission[idx].Key, r.Perm.Key) && bytes.Equal(role.KeyPermission[idx].RangeEnd, r.Perm.RangeEnd) {
+		if role.KeyPermission[idx].PermType == r.Perm.PermType {
+			if as.lg != nil {
```

Comment:
I don't think we need to check for nil lg here?

Reply:
yes. the logger will be validated when starting etcd server.